SourcetrailExtension/SolutionParser/CompilationDatabase.cs

```diff
@@ -137,5 +137,26 @@ public void Clear()
 		{
 			_compileCommands.Clear();
 		}
+
+		public override bool Equals(object obj)
+		{
+			var database = obj as CompilationDatabase;
+			if (ReferenceEquals(this, database))
```

Comment:
The only difference from the VS-generated `Equals` is the addition of this ReferenceEquals check, because operator== has one.